lib/galaxy/jobs/__init__.py

```diff
@@ -422,8 +422,10 @@ def __get_params(self, parent):
             key = param.get('id')
             if key in ["container", "container_override"]:
                 from galaxy.tools.deps import requirements
+                # Get container elements contained within the param.
+                # https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.getchildren
                 containers = map(requirements.container_from_element, list(param))
```

Comment:
Can you `s|list(param)|param.findall('container')|` instead? It's more specific and seems to work for me.

Reply:
That is much better - thanks!